### PR TITLE
Audit VIconButton sizes and props

### LINKED ISSUE #1894: Add accessible button components for mobile searchbar
## Problem

The mobile search bar in the new header contains buttons with focus area of 32x32px which need to have tappable area of more than 40x40px to be accessible.

## Description

Create a new component for buttons based on the `VIconButton` component that is 48x48px in size, but has a visible focus area of 32x32px to match the designs.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1894 by @obulat

## Description

This PR is an alternative to #1895. It keeps using the `size` prop because otherwise, we would have to use classes like `!w-4` to override the default icon size, or not have a default icon size and add size classes to _every_ VIcon..

This PR audits all the `VIconButton`s:
- The number of sizes is now down to 6:
```
  tiny: { icon: 6, button: 'w-6 h-6' },
  small: { icon: 6, button: 'w-10 h-10' },
  medium: { icon: 6, button: 'w-12 h-12' },
  larger: { icon: 8, button: 'w-14 h-14' },
  large: { icon: 12, button: 'w-20 h-20' },
```
There is a Storybook VR test for `VIconButton` sizes.
- Unnecessary `button-props` declarations are re-written. There's no need to pass `aria-label` or `variant="plain"` to `button-props` because the first can be passed directly to the `VIconButton`, and the second is the default.

This PR also adds a Storybook story for VSearchBarButton; replaces `VClearButton` with a `VSearchBarButton` in the desktop search bar.

## Testing Instructions

The CI should pass, including all the audio VR tests that use various sizes of `VIconButton`, and all of the Storybook tests.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dhruvkb — APPROVED** [review]
Did you miss the tests? Can't see them in the PR. All good otherwise!

File: src/components/VIconButton/VIconButton.vue [hunk 1]
```diff
@@ -26,23 +26,11 @@ import VIcon, { type IconProps } from '~/components/VIcon/VIcon.vue'
...
-    icon: 'w-6 h-6',
-    button: 'w-14 md:w-[69px] h-14 md:h-[69px]',
-  },
-  medium: { icon: 'w-8 h-8', button: 'w-14 h-14' },
-  large: { icon: 'w-12 h-12', button: 'w-20 h-20' },
+  tiny: { icon: 6, button: 'w-6 h-6' },
+  small: { icon: 6, button: 'w-10 h-10' },
+  medium: { icon: 6, button: 'w-12 h-12' },
+  larger: { icon: 8, button: 'w-14 h-14' },
+  large: { icon: 12, button: 'w-20 h-20' },
```
Comment (sarayourfriend) [inline src/components/VIconButton/VIconButton.vue:33]: Just a nit pick on the names here: In my head I expected `larger` to be bigger than `large` :thinking: Maybe `larger -> large` and `large -> extra-large` or something would be clearer in the "order" of how big one or the other is?

**Review by sarayourfriend — APPROVED** [review]
Love the tests and really appreciate how much confidence the visual regression tests give us when working on these things. Kudos to everyone who's been writing more and more of them :rocket: 

The changes LGTM. I left just one non-blocking note about the names of the sizes.

File: src/components/VIconButton/VIconButton.vue (hunk 1 above)
Comment (obulat) [reply src/components/VIconButton/VIconButton.vue:33]: Thank you for helping with the naming :)

**Pull request merged**